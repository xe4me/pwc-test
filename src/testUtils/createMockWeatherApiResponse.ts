export const createMockWeatherApiResponse = (city_name: string) => ({
    city_name,
    data: [
      {
        wind_cdir: 'W',
        pres: 1014,
        high_temp: 20.2,
        wind_gust_spd: 5.9,
        wind_spd: 3.1,
        dewpt: 5.9,
        weather: {
          icon: 'c02d',
          code: 801,
          description: 'Few clouds',
        },
        wind_dir: 266,
        precip: 0,
        low_temp: 9.3,
        max_temp: 20.2,
        datetime: '2022-06-25',
        temp: 14.9,
        min_temp: 9.9,
      },
    ],
  });
