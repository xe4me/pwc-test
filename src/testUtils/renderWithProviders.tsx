import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { ThemeProvider } from '@mui/material';
import theme from '../theme';

export const renderWithProviders = (element: ReactElement) =>
  render(<ThemeProvider theme={theme}>{element}</ThemeProvider>);
