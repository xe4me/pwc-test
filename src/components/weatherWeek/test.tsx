import React from 'react';
import { renderWithProviders } from 'testUtils/renderWithProviders';
import { axe } from 'jest-axe';
import { createMockWeatherApiResponse } from 'testUtils/createMockWeatherApiResponse';
import { WeatherWeek } from './index';

const renderComponent = () => renderWithProviders(
    <WeatherWeek daysForecasts={createMockWeatherApiResponse('Sydney').data} />
  );
describe('<WeatherWeek>', () => {
  describe('Snapshot and accessibility', () => {
    it('should match with the snapshot', () => {
      const { baseElement } = renderComponent();
      expect(baseElement).toMatchSnapshot();
    });
    it('should pass accessibility tests', async () => {
      const html = renderComponent();
      expect(await axe(html.container)).toHaveNoViolations();
    });
  });
});
