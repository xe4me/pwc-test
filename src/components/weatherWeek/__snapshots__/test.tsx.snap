// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherWeek> Snapshot and accessibility should match with the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-true css-fy6j78-MuiGrid-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-subtitle1 css-1isisa7-MuiTypography-root"
        >
          Today
        </h4>
        <img
          alt="Few clouds"
          class="css-g8qt4o"
          src="https://www.weatherbit.io/static/img/icons/c02d.png"
        />
        <table>
          <tbody>
            <tr>
              <td>
                <div
                  class="css-1t3828c"
                >
                  <h4
                    class="MuiTypography-root MuiTypography-h6 css-1a0934u-MuiTypography-root"
                  >
                    <span>
                      20.2
                    </span>
                    <span
                      class="css-wu6n3n"
                    >
                      ○
                    </span>
                  </h4>
                   
                  <h4
                    class="MuiTypography-root MuiTypography-h6 css-1wepp2i-MuiTypography-root"
                  >
                    <span>
                      9.3
                    </span>
                    <span
                      class="css-wu6n3n"
                    >
                      ○
                    </span>
                  </h4>
                </div>
              </td>
            </tr>
            <tr>
              <td>
                <h4
                  class="MuiTypography-root MuiTypography-subtitle2 css-1nttt8y-MuiTypography-root"
                >
                  <div
                    class="css-1t3828c"
                  >
                    <span>
                      Pollen
                    </span>
                    <span>
                      14.9
                    </span>
                  </div>
                </h4>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</body>
`;
